### PR TITLE
Revert "ENH make sure each compiler version gets a CI job"

### PR DESCRIPTION
**Pull request by beckermr**
Reverts conda-forge/conda-smithy#1405

### CODE REVIEW

**Comment by beckermr** [comment]
cc @isuruf

File: news/TEMPLATE.rst [hunk 1]
```diff
@@ -21,3 +21,4 @@
 **Security:**
 
 * <news item>
+
```
Comment (beckermr) [inline news/TEMPLATE.rst:24]:
```suggestion
```

**Comment by isuruf** [comment]
Thanks

**Comment by beckermr** [comment]
Feel free to release whenever.

**Comment by jakirkham** [comment]
Why did it need to be reverted?

**Comment by beckermr** [comment]
The pr was fine, but we found the actual bug in conda-build. I put a workaround in the mpich feedstock until the conda-build bug gets fixed. See here: https://github.com/conda/conda-build/issues/4090

**Comment by jakirkham** [comment]
Ah ok. Thanks for the context 🙂

**Pull request merged**